main-settings/src/main/scala/sbt/Structure.scala

```diff
@@ -462,14 +462,17 @@ object Scoped {
   ) {
 
     type App[T] = Initialize[Task[T]]
+
+    /** A higher-kinded function, where each parameter shares the same type constructor `M[_]`. */
     type Fun[M[_], Ret]
 
+    /** Convert the higher-kinded function to a Function1.  For tuples that means call `.tupled`. */
     protected def convert[M[_], Ret](f: Fun[M, Ret]): K[M] => Ret
 
     private[this] val inputs: K[App] = a.transform(keys, λ[ScopedTaskable ~> App](_.toTask))
 
     private[this] def onTasks[T](f: K[Task] => Task[T]): App[T] =
-      Def.app[λ[L[x] => K[(L ∙ Task)#l]], Task[T]](inputs)(f)(AList.asplit[K, Task](a))
+      Def.app[AList.Split[K, Task]#l, Task[T]](inputs)(f)(AList.asplit[K, Task](a))
```

Comment:
Not sure if this change makes the code more readable.

Reply:
Do you mean this use site, or the introduction of `Split` as a whole?

The reason I introduced it is to avoid having to do any of the alternatives to define this type parameter (and you _have_ to define the type parameter, as the type inferencer will never infer this type parameter), as in: use kind-projector, use Jason's original anonymous type lambda thing, or define your own type alias and then use it.

It follows the lead set by tuple2/tuple3/etc which have the `T2K`/`T3K`/etc helpers.

Reply:
> Do you mean this use site, or the introduction of `Split` as a whole?

`Split` as a whole I guess. Isn't kind projector the current semi-standard notation on these weird type things?

Reply:
That's a little bit true.  IntelliJ kind of "supports" it now.  But Dotty only implements parts of it, and with some changes to syntax.

But, anyways, kind projector is for defining inline type lambda.  You can still use aliases, and that might be clearer.  (It's as clear and unclear as other forms of type aliases are.)